documentation/api/api-versioning.md

```diff
@@ -0,0 +1,56 @@
+# Overview
+
+This document will detail how we handle versioning our API right now, as well a long-term idea that we will want to circle back to.
+
+## Why versioning?
+
+While many changes we plan to make to our API should be backwards-compatible, there will be times when we have to make
+changes that would break our clients from calling our API (eg. making a field required that was previously optional).
+
+By having versioned endpoints, we can give our clients time to update their calls until the old version of the endpoint can be fully deprecated.
+
+# Current Implementation
+
+Endpoints are versioned in their routes. For example, if our `GET` opportunity endpoint is `GET /v1/opportunities/:opportunityId` and
+we made a change that was not backwards compatible (let's say we change the authentication approach), then we'd make an entirely
+new endpoint called `GET /v2/opportunities/:opportunityId`. The `v1` endpoint would still exist, and most of the logic would likely be shared
+between the implementations, but they would be two entirely separate endpoints as far as API routing is concerned.
+
+While we are in the early phases of developing the API, prior to having any significant users of the API, we won't worry about
+maintaining the backwards compatability. We won't start adding new versions of the API until we would begin impacting production
+systems of our users.
+
+# Long-term idea using headers
+
+An alternative approach we could take in the future is to instead put the versioning in a header parameter (like the `content-type` field) similar to [Stripe](https://stripe.com/blog/api-versioning).
+
+However, that approach has a few complexities due to the libraries we use for defining schemas and endpoints:
+
+* [webargs](https://webargs.readthedocs.io/en/latest/index.html) - a generic framework for parsing HTTP requests, and can serialize/deserialize using Marshmallow schemas.
+* [apispec](https://apispec.readthedocs.io/en/latest/) - a tool that can generate OpenAPI specifications from Marshmallow schemas.
+* [APIFlask](https://apiflask.com/) - the framework we use that connects Flask, webargs, and apispec together for us.
```

Comment:
This is helpful callout, especially since these libraries provide us with valuable abstractions that we _do_ want to preserve, namely auto-generating the OpenAPI spec